test/integration/connection.test.js

```diff
@@ -0,0 +1,29 @@
+import { expect } from 'chai';
+import BeanstalkdWorker from '../../src/index';
+
+describe('Connection', () => {
+  it('should throw an error when it\'s not able to connect to the queue', async function () {
+    this.timeout(10000);
+
+    const promise = new Promise((resolve, reject) => {
+      const worker = new BeanstalkdWorker('127.0.0.1', '65534', {
+        onConnectionError: reject,
```

Comment:
we'd probably want to stop watching on connection error as well?
can you please also update the documentation with an example of how to bail out if it continues being unable to connect.

Reply:
> we'd probably want to stop watching on connection error as well?

I'm not completely sure. When we can't connect, we'll try again in 500ms and see if the connection is up. If we'd bail out, this would stop the service completely from watching. (this would also be a breaking change).

Reply:
What we intended to do internally, was to add some logic to the

Reply:
> What we intended to do internally, was to add some logic to the onConnectionError to stop the worker/let our service exit completely

I was suggesting we do this in the test, not automatically on connection error, but as an example of how to bail out :)

Reply:
Oh right, I'll try to add a test for this